lang/javascript/javascript.talon

```diff
@@ -38,14 +40,9 @@ state async: "async "
 
 state await: "await "
 
-state map:
-    user.insert_between(".map(", ")")
+state {user.code_chain_function}:
+    user.insert_between(".{code_chain_function}(", ")")
 
-state filter:
-    user.insert_between(".filter(", ")")
-
-state reduce:
-    user.insert_between(".reduce(", ")")
```

Comment:
Consider outputting a deprecation warning on the use of `state` here then add `chain` for user.code_chain_function.